### PR TITLE
Add version args to base-notebook dockerfile

### PR DESCRIPTION
**Pull request by asdfsx**
To install specific notebook/jupyterhub/jupyterlab into base-notebook image, add version args to dockerfile

### CODE REVIEW

File: base-notebook/Dockerfile [hunk 1]
```diff
@@ -126,16 +126,19 @@ RUN set -x && \
             fix-permissions "/home/${NB_USER}"; \
     fi;
 
+ARG NOTEBOOK_VERSION=default
+ARG JUPYTERHUB_VERSION=default
+ARG JUPYTERLAB_VERSION=default
```
Comment (mathbunnyru) [inline base-notebook/Dockerfile:131]: If we change default values to empty strings, then the check is not needed and we can just write
```
mamba install --quiet --yes notebook="${NOTEBOOK_VERSION}"
```

And if someone specified version, they will get it and if not - they will get default version.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

**Comment by consideRatio** [comment]
This is a slippery slope, why not allow configuration of another software version as well if we allow these three to be configurable?

I suggest to reject thus PR, the workaround of building from an already built image and installing a specific version doesnt seem to cause much downside to me.

Another option would be to perhaps declare version restrictions in conda before it resolves things, then the user could provide any kind of restriction without us having to have logic hardcoded for individual packages etc.

**Comment by mathbunnyru** [comment]
What @consideRatio tells makes a lot of sense to me.
Both his options seem reasonable and building from an existing image in my opinion is much easier than rebuilding the whole stack with some environment variables set.

**Comment by asdfsx** [comment]
I'm trying to build a base-notebook image with specific version.   
And I found there is a variable to specific PYTHON version, so I add these 3 variables.

Maybe I should just using the existing notebook and reinstall jupyterlab and jupyterhub with other version.

**Comment by mathbunnyru** [comment]
Yes, it would be a better choice.

**Pull request closed without merging**